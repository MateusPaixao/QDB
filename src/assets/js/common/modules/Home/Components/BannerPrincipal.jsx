const Methods = {
    BuildBanners(){
        class BannerPrincipal extends React.Component {
            constructor(props){
                super(props);
                this.state = {
                    BannersImgs: [],
                    BannersUrls: [],
                    Viewport: "Desktop"
                }
                this.getBanners = this.getBanners.bind(this);
                this.setViewport = this.setViewport.bind(this);
            }

            setViewport(size){
                let actualViewport = size <= 768 ? "Mobile" : "Desktop"
                return(
                    this.setState({
                        Viewport: actualViewport
                    })
                )
            }

            getBanners(){
                let bannerViewport = document.querySelector(".banners" + this.state.Viewport);
                for(let i = 1; i < bannerViewport.textContent.split('<div class="box-banner">').length; i++){
                    this.state.BannersImgs.push(bannerViewport.textContent.split('<div class="box-banner">')[i].match(/src\s*=\s*"(.+?)"/)[1]);
                    console.log(bannerViewport.textContent.split('<div class="box-banner">')[i].match(/src\s*=\s*"(.+?)"/)[1]);
                    this.state.BannersUrls.push(bannerViewport.textContent.split('<div class="box-banner">')[i].match(/href\s*=\s*"(.+?)"/)[1]);
                    
                    // let idImg = string.match(/\d+/);;
                    // string.replace(idImg[0], idImg[0] + "#{width}-#{height}")
                }
            }
            
            // handleChange(e){
            //   this.setState({

            //   })
            // }
        
            componentDidMount(){
                this.setViewport(window.innerWidth);
                this.getBanners();
            }
            
            render(){
                let banners = [];
                const Banners = () => {
                    this.state.BannersImgs.map((banner, i)=>{
                        banners.push(
                            <a className="bannerHero__link" href={this.state.BannersUrls[i]} key={i}>
                                <img className="bannerHero__img" src={banner} loading="lazy" />
                            </a>
                        )
                    });

                    return banners;
                }

                return (
                    <React.Fragment>
                        <Banners />
                    </React.Fragment>
                )
            }
        
        }
        
        ReactDOM.render(
            <BannerPrincipal />,
            document.getElementById('bannerHero--render')
        );
    }
}
  
export default {
    init: Methods.BuildBanners
};