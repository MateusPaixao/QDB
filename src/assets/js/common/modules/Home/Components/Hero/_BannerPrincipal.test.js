import React from 'react';
import BannerPrincipal from './_BannerPrincipal';

describe('<Home/>', () => {
  it('renders and displays information properly', () => {
      const wrapper = mount(
          <BannerPrincipal Url="https://www.quemdisseberenice.com.br/box-surpresa_20200301/p?idsku=2632" Src="https://qbbr.vteximg.com.br/arquivos/ids/171097/qdb_c3_4_GOTAS-SECANTES.png" />
      );
      // console.log(wrapper.debug());
      expect(wrapper.find('[dataid="banner"]')).toMatchSnapshot();
  });
});
