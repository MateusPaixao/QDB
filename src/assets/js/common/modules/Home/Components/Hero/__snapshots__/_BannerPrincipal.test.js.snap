// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders and displays information properly 1`] = `
<a
  className="bannerHero__link"
  dataid="banner"
  href="https://www.quemdisseberenice.com.br/box-surpresa_20200301/p?idsku=2632"
>
  <img
    className="bannerHero__img"
    data-src="https://qbbr.vteximg.com.br/arquivos/ids/171097/qdb_c3_4_GOTAS-SECANTES.png"
    loading="lazy"
  />
</a>
`;
